src/main/java/com/microsoft/sqlserver/jdbc/SQLServerBlob.java

```diff
@@ -143,6 +149,11 @@ private void checkClosed() throws SQLServerException {
     @Override
     public InputStream getBinaryStream() throws SQLException {
         checkClosed();
+        // If the LOB is currently streaming and the stream hasn't been read, read it.
+        if (null == value && !activeStreams.isEmpty() && loadLob) {
```

Comment:
I think a minor improvement here would be consolidate the two if statements that are on line 153 and 157 into one, then have a nested if statement that only checks for `loadLob` since the other two conditions are checked twice.

Reply:
Also applies to the SQLServerClob file.

Reply:
Can't be done, the point is to do this before the lobs create the streams, if we nest the if's, it'll result in the lobs loading the stream into a string and then attempting to return a streaming inputstream.